### PR TITLE
derived constant in generator vue

### PR DESCRIPTION
**Pull request by Tcharl**
Contributes to #14235
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: utils/entity.js [hunk 1]
```diff
@@ -357,6 +357,7 @@ function prepareEntityPrimaryKeyForTemplates(entityWithConfig, generator, enable
       nameCapitalized: _.upperFirst(primaryKeyName),
       type: primaryKeyType,
       tsType: generator.getTypescriptKeyType(primaryKeyType),
+      isNumeric: !composite && (primaryKeyType === LONG || primaryKeyType === INTEGER),
```
Comment (mshima) [inline utils/entity.js:360]:
```suggestion
```

File: utils/entity.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ const { MapperTypes } = require('../jdl/jhipster/entity-options');
 const { OAUTH2 } = require('../jdl/jhipster/authentication-types');
 const { CommonDBTypes } = require('../jdl/jhipster/field-types');
 
-const { BOOLEAN } = CommonDBTypes;
+const { BOOLEAN, LONG, INTEGER } = CommonDBTypes;
```
Comment (mshima) [inline utils/entity.js:33]:
```suggestion
const { BOOLEAN } = CommonDBTypes;
```

File: generators/entity-client/templates/vue/src/test/javascript/spec/app/entities/entity.service.spec.ts.ejs [hunk 3]
```diff
@@ -57,55 +57,54 @@ describe('Service Tests', () => {
...
     beforeEach(() => {
       service = new <%= entityAngularName %>Service();
-      <%_ if (fieldsContainDate) { _%>
+<%_ if (fieldsContainDate) { _%>
       currentDate = new Date();
-      <%_ } _%>
+<%_ } _%>
       elemDefault = new <%= entityAngularName %>(
-        <%_ if (primaryKey.type === 'String'  || primaryKey.type === 'UUID') { _%>
+<%_ if (!primaryKey.isNumeric) { _%>
```
Comment (mshima) [inline generators/entity-client/templates/vue/src/test/javascript/spec/app/entities/entity.service.spec.ts.ejs:70]:
```suggestion
        <%- generateTestEntityId(primaryKey.type) %>
```
The entire condition

File: generators/entity-client/templates/vue/src/test/javascript/spec/app/entities/entity.service.spec.ts.ejs [hunk 4]
```diff
@@ -128,18 +127,17 @@ describe('Service Tests', () => {
 <%_ if (!readOnly) { _%>
       it('should create a <%= entityAngularName %>', async () => {
         const returnedFromService = Object.assign({
-  <%_ if (primaryKey.type === 'String'  || primaryKey.type === 'UUID') { _%>
+  <%_ if (!primaryKey.isNumeric) { _%>
```
Comment (mshima) [inline generators/entity-client/templates/vue/src/test/javascript/spec/app/entities/entity.service.spec.ts.ejs:130]:
```suggestion
          <%- generateTestEntityId(primaryKey.type) %>
```
The entire condition

**Comment by Tcharl** [comment]
So @mshima it's considered reviewed?

**Comment by mshima** [comment]
@Tcharl I didn’t review everything, but changes to source are only id sample and line removals.
I think it’s good to go.

**Pull request merged**